### ISSUE TITLE
Ifpack2 FastIC&FastILDL: Crash when using non-default node type

### ISSUE THREAD

**Issue by cgcgcg** [trilinos/Trilinos#12685]
## Bug Report
PR #12678 adds a test that checks that symbolic and numeric phases are respected for Ifpack2 preconditioners. As part of the new test, I ran FastIC and FastILDL on `Serial` node in a build with default node type `Cuda`. This results in 
```
Kokkos::View ERROR: attempt to access inaccessible memory space (label="**UNAVAILABLE**")` failed.
```
in preconditioner apply.

**Comment by cgcgcg** [comment]
@jgfouca Are you still working on the Fast* methods? If so, would you be able to have a look at this?

**Comment by jgfouca** [comment]
@cgcgcg , I worked on FastILU, not FastIC or FastLDL. That said, I could probably figure this out if @srajama1 wants me to look at this. I will check with him.

**Comment by jgfouca** [comment]
@cgcgcg , I will take on this issue. Could you give me a bit more details on how to reproduce the problem? I assume I need your cgcgcg:ifpack2symbolicnumeric branch. 

My cmake command (this is on weaver):

```
cmake -DCMAKE_BUILD_TYPE=Release -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_ShyLU=On -DTrilinos_ENABLE_ShyLU_NodeFastILU=On -DTrilinos_ENABLE_Ifpack2=On -DTrilinos_ENABLE_ShyLU_Node=On -DTrilinos_ENABLE_ShyLU_DD=On -DTrilinos_ENABLE_ShyLU_DDFROSch=On -DTPL_ENABLE_MPI=ON -DTPL_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DTpetra_INST_SERIAL=On ..
```

**Comment by cgcgcg** [comment]
All it should take is to run `packages/ifpack2/test/unit_tests/Ifpack2_SolverFactory.exe --details=ALL`.

**Comment by cgcgcg** [comment]
Fixed by #12703. Thanks @jgfouca!
